src/main/java/com/aws/greengrass/builtin/services/telemetry/ComponentMetricIPCEventStreamAgent.java

```diff
@@ -0,0 +1,178 @@
+    }
+
+    class PutComponentMetricOperationHandler extends GeneratedAbstractPutComponentMetricOperationHandler {
+        private final String serviceName;
+
+        protected PutComponentMetricOperationHandler(OperationContinuationHandlerContext context) {
+            super(context);
+            serviceName = context.getAuthenticationData().getIdentityLabel();
+        }
+
+        @Override
+        protected void onStreamClosed() {
+            // NA
+        }
+
+        @SuppressWarnings({"PMD.PreserveStackTrace", "PMD.AvoidCatchingGenericException"})
+        @Override
+        public PutComponentMetricResponse handleRequest(PutComponentMetricRequest componentMetricRequest) {
+            return translateExceptions(() -> {
+                logger.atDebug().log("Received putComponentMetricRequest from component " + serviceName);
+
+                // Authorize request for given operation
+                try {
+                    doAuthorization(this.getOperationModelContext().getOperationName(), serviceName);
+                } catch (AuthorizationException e) {
+                    throw new UnauthorizedError(e.getMessage());
+                }
+
+                //validate - metric name length, value is non negative etc etc
+                List<software.amazon.awssdk.aws.greengrass.model.Metric> metricList =
+                        componentMetricRequest.getMetrics();
+                try {
+                    validateComponentMetricRequest(metricList);
+                } catch (IllegalArgumentException e) {
+                    throw new InvalidArgumentsError(e.getMessage());
+                }
+
+                // Perform translations on metrics List
+                try {
+                    final String metricNamespace = serviceName;
+                    translateAndEmit(metricList, metricNamespace);
+                } catch (IllegalArgumentException e) {
+                    throw new InvalidArgumentsError(e.getMessage());
+                } catch (Exception ex) {
+                    throw new ServiceError(ex.getMessage());
+                }
+
+                return new PutComponentMetricResponse();
+            });
+        }
+
+        @Override
+        public void handleStreamEvent(EventStreamJsonMessage streamRequestEvent) {
+            // NA
+        }
+    }
+
+    // Translate request metrics to telemetry metrics and emit them
+    private void translateAndEmit(List<software.amazon.awssdk.aws.greengrass.model.Metric> componentMetrics,
+                                  String metricNamespace) {
+        final MetricFactory metricFactory =
+                metricFactoryMap.computeIfAbsent(metricNamespace, k -> new MetricFactory(metricNamespace));
+
+        componentMetrics.forEach(metric -> {
+            logger.atDebug().log("Translating component metric to Telemetry metric" + metric.getName());
+            Metric telemetryMetric = getTelemetryMetric(metric, metricNamespace);
```

Comment:
You can just use the `builder` here instead of having another function.

Reply:
Mainly did this for readability, but I'm okay adding the code here if you feel strongly about it.

Reply:
I don't think we should have a function for a one liner. but I don't feel strongly about it. So I'll leave it up to you.